src/backend/drm/impl/Atomic.cpp

```diff
@@ -101,8 +101,8 @@ void Aquamarine::CDRMAtomicRequest::addConnector(Hyprutils::Memory::CSharedPoint
     add(connector->crtc->id, connector->crtc->props.active, enable);
 
     if (enable) {
-        if (connector->output->supportsExplicit && STATE.explicitOutFence >= 0)
-            add(connector->crtc->id, connector->crtc->props.out_fence_ptr, STATE.explicitOutFence);
+        if (connector->output->supportsExplicit && STATE.committed & COutputState::AQ_OUTPUT_STATE_EXPLICIT_OUT_FENCE)
+            add(connector->crtc->id, connector->crtc->props.out_fence_ptr, (uintptr_t)&STATE.explicitOutFence);
```

Comment:
why are you committing the out fence if you just commented out the setter? maybe not comment the setter out but this instead?

Reply:
It commits the pointer to a value so it can be written by something external. At least I understand it this way. That's what wlroots is doing. The setter only signals that this pointer is needed when some timeline is used.

Reply:
OH, alright.

Reply:
Make it not cleared on ::commit then maybe in COutputState? that clears it

Reply:
wlroots clears it in `drm_connector_state_init`. I guess it is relevant across one frame only. If this `drm_connector_state_init` is called every frame. I am confused %)

Reply:
IIRC it is. You can clear it in the CDRMOutput::commit just before impl->commit() then

Reply:
wlroots set the timeline and uses it  in `drm_atomic_connector_prepare` to get `primary_in_fence_fd`. aquamarine gets `explicitInFence` directly from hl. Clearing it before impl->commit() would be too early. I think it is fine to leave it as is.

Reply:
but now hyprland can't access out because it's cleared instantly after connector->commitState on L1379
![image](https://github.com/hyprwm/aquamarine/assets/43317083/865df4be-e0e4-44c8-a318-baa6b8062404)

state->onCommit clears the outFence

Reply:
hmm... looks like with the current api hl should do the clearing